src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -205,15 +207,18 @@ Value insertAllocOrEmitZeroConstant(ArrayRef<IndexExpr> dims,
             /*value=*/nullptr,
             /*alignment=*/rewriter.getI64IntegerAttr(4096));
 
-    // Use an opaque attribute to store stickified data.
+    // Use an dense resource attribute to store stickified data.
     // Attribute type: tensor<sizeInBytes x i8>
-    int64_t sizeInBytes = getMemRefSizeInBytes(resType).getValue();
+    int64_t sizeInBytes = getMemRefSizeInBytes(resType).value();
     char *rawData = (char *)malloc(sizeInBytes);
     memset(rawData, 0, sizeInBytes);
-    OpaqueElementsAttr valueAttr =
-        OpaqueElementsAttr::get(stickifiedConstant.getOperation()->getDialect(),
-            RankedTensorType::get({sizeInBytes}, rewriter.getI8Type()),
-            StringRef(rawData, sizeInBytes));
+    DenseResourceElementsAttr valueAttr = DenseUI8ResourceElementsAttr::get(
+        RankedTensorType::get({sizeInBytes}, rewriter.getI8Type()),
+        stickifiedConstant.getOperation()
+            ->getDialect()
+            ->getNamespace(), // use the dialect as the blob "hint"
+        HeapAsmResourceBlob::allocateAndCopy(
+            ArrayRef(rawData, sizeInBytes), alignof(char)));
```

Comment:
Thanks for this change! 

Heap data is mutable by default:
```C
AsmResourceBlob allocateAndCopy(ArrayRef<char> data, size_t align, bool dataIsMutable = true)
```
I would set `dataIsMutable =false` since we store constants (weights) and do not expect any modification.

Reply:
Done

Reply:
@tungld : I had to revert this change. The data cannot be marked as immutable if we want to use `allocateAndCopy`. I think this is an issue with its implementation and it may be worth updating in `mlir` but unless/until it happens, it should continue to be marked as mutable.

```
  /// Create a new heap allocated blob and copy the provided data into it.
  static AsmResourceBlob allocateAndCopy(ArrayRef<char> data, size_t align,
                                         bool dataIsMutable = true) {
    AsmResourceBlob blob = allocate(data.size(), align, dataIsMutable);
    **std::memcpy(blob.getMutableData().data(), data.data(), data.size());**
    return blob;
  }
```

Reply:
I see. Our NNPA machine is down now and I haven't had chance to check the current code again. Other than that, the code looks good to me. I hope the machine will be back in one or two days and that you don't mind to wait.